animations/lib/src/basics/08_animated_list.dart

```diff
@@ -0,0 +1,137 @@
+// Copyright 2020 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math';
+
+import 'package:flutter/material.dart';
+import 'package:flutter/widgets.dart';
+
+class AnimatedListDemo extends StatefulWidget {
+  static String routeName = '/basics/08_animated_list';
+
+  _AnimatedListDemoState createState() => _AnimatedListDemoState();
+}
+
+class _AnimatedListDemoState extends State<AnimatedListDemo> {
```

Comment:
This state object is holding the GlobalKey, but no other actual state. It needs to contain the "master" copy of the list of people. You can fix this by:

* Adding a `listData` field that holds a list of user models
* Initializing it with the data from the current listData array below (try renaming that one to `initialListData` or something similar).
* When the code adds or removes an item, update the list held by this stateful widget and *then* call insert or remove on the AnimatedListState. These can be done in a single `setState` call.

Reply:
@RedBrogdon Thanks for the review. 😃
I have made all other requested changes but I am not fully sure that I have made this particular change as requested. Pls review if something more need to be changed.